sympy/integrals/transforms.py

```diff
@@ -1826,19 +1767,88 @@ def _collapse_extra(self, extra):
         return plane, cond
 
     def _try_directly(self, **hints):
-        fn = self.function
-        debug('----> _try_directly: %s'%(fn, ))
-        t_ = self.function_variable
-        s_ = self.transform_variable
-        LT = None
-        if not fn.is_Add:
-            fn = expand_mul(fn)
+        T = None
+        try_directly = not any(func.has(self.function_variable)
+                               for func in self.function.atoms(AppliedUndef))
+        if try_directly:
```

Comment:
Is this a duplication of what is in doit?

Reply:
No, it is is a duplicate of the `_try_directly` of the `IntegralTransform` without the lines 

```
        if not fn.is_Add:
            fn = expand_mul(fn)
```

However, my new code does not use its own `_try_directly` method anymore, so I can just as well remove it from the code. Change (N).